lsp-mode.el

```diff
@@ -4277,8 +4279,8 @@ Also, additional data to attached to each candidate can be passed via PLIST."
                                  it))))
                   (-flatten-n 1)
                   (-sort (-on #'> (lambda (o)
-                                    (or (get-text-property 0 'completion-score o)
-                                        0))))
+                                    (* (or (get-text-property 0 'completion-score o) 1.0)
+                                       (or (get-text-property 0 'lsp-completion-score o) 0)))))
```

Comment:
seems like the second 0 should be 1.0

Reply:
Ah, correct, I've changed my mind, setting to a small number as penalty in case the score is not existed instead

Reply:
I am a bit concerned that these calculations are on the hot path(the completion) and they will be calculated N*LogN during the sorting. Probably, a bit more efficient will be to calculate the scores one time and use them during the sorting. Some language servers like dart do not filter and have 1000+(?) candidates. Another option is to reorganize the code so we use the old comparison function when the first candidate does not have 'score. WDYT?

Reply:
I see. While I'm expecting calculating the multiply can be quite fast since it's implemented in C code, calculating score just one seems to be better optimization.
I've fixed it by caching the calculated value.

Reply:
The multiplication is fast but the function call elisp->c is slow. So you have 2 `or`'s one `get` and one `*` and this one will be repeated N*LogN while the user is typing.